graphdatascience/query_runner/cypher_graph_constructor.py

```diff
@@ -73,51 +71,244 @@ def _should_warn_about_arrow_missing(self) -> bool:
+
+            return GraphColumnSchema(node_schema, rel_schema)
+
+        def adjust_node_df(self, node_df: DataFrame, rel_df: DataFrame, schema: GraphColumnSchema) -> DataFrame:
+            node_dict: Dict[str, Any] = {
+                "sourceNodeId": node_df["nodeId"],
+                f"targetNodeId{self._BIT_COL_SUFFIX}": False,
+                "targetNodeId": -1,
+                "relationshipType": None,
+            }
+
+            if "labels" in schema.nodes.all:
+                node_dict["sourceNodeLabels"] = node_df["labels"]
+
+            # as we first list all nodes at the top of the df, we dont need to lookup properties for the target node
+            for col in schema.nodes.properties:
+                node_dict[col + self._BIT_COL_SUFFIX] = True
+                node_dict[col] = node_df[col]
+
+            for col in schema.relationships.properties:
+                node_dict[col + self._BIT_COL_SUFFIX] = False
+                dummy_value = rel_df[col][0]
+                node_dict[col] = dummy_value if not type(dummy_value) is list else [[] for i in range(0, len(node_df))]
+
+            return DataFrame(node_dict)
+
+        def adjust_rel_df(self, rel_df: DataFrame, node_df: DataFrame, schema: GraphColumnSchema) -> DataFrame:
+            rel_dict = {
+                "sourceNodeId": rel_df["sourceNodeId"],
+                "targetNodeId": rel_df["targetNodeId"],
+                f"targetNodeId{self._BIT_COL_SUFFIX}": True,
+            }
+
+            if "relationshipType" in schema.relationships.all:
+                rel_dict["relationshipType"] = rel_df["relationshipType"]
+
+            if "labels" in schema.nodes.all:
+                rel_dict["sourceNodeLabels"] = None
+
+            for col in schema.relationships.properties:
+                rel_dict[col + self._BIT_COL_SUFFIX] = True
+                rel_dict[col] = rel_df[col]
+
+            for col in schema.nodes.properties:
+                rel_dict[col + self._BIT_COL_SUFFIX] = False
+                dummy_value = node_df[col][0]
+                rel_dict[col] = dummy_value if not type(dummy_value) is list else [[] for i in range(0, len(rel_df))]
+
+            return DataFrame(rel_dict)
+
+        def nodes_config(self, combined_cols: List[str], node_cols: Set[str]) -> str:
+            # Cannot use a dictionary as we need to refer to the `data` variable in the cypher query.
+            # Otherwise we would just pass a string such as `data[0]`
+            nodes_config_fields: List[str] = []
+
+            if "labels" in node_cols:
+                nodes_config_fields.append(f"sourceNodeLabels: data[{combined_cols.index('sourceNodeLabels')}]")
+
+            property_columns: List[str] = list(set(node_cols) - {"nodeId", "labels"})
+            property_columns.sort()
+
+            # as we first list all nodes at the top of the df, we dont need to lookup properties for the target node
+            if len(property_columns) > 0:
```

Comment:
I believe some linter would suggest to use 

```suggestion
            if property_columns:
```

as empty lists/sets/anything are falsey